axiom/datasets_integration_test.go

```diff
@@ -327,37 +327,57 @@ func (s *DatasetsTestSuite) TestCursor() {
 	// retrieved from the query status MinCursor or MaxCursor fields, depending
 	// on the queries sort order.
 	midRowID := queryResult.Matches[1].RowID
-	midTime := queryResult.Matches[1].Time
 
-	// Query events with a cursor in descending order.
+	// Query events with a cursor in descending order...
 	apl = fmt.Sprintf("['%s'] | sort by _time desc", s.dataset.ID)
 	queryResult, err = s.client.Datasets.Query(s.ctx, apl,
-		// HINT(lukasmalkmus): Start time is inclusive when sorting newest
-		// first.
-		query.SetStartTime(midTime.Add(-time.Nanosecond)),
+		query.SetStartTime(startTime),
+		query.SetEndTime(endTime),
```

Comment:
Just to understand what's going on here: the rowID here is actually the full encoded cursor with timestamp and "real" rowID? So server side we end up overwriting the time boundaries from it?

In normal "low level usage" I think the start time should also be set from the row that we grab the rowID from -- so even though it was clunky before, I think it was also more correct -- but as I suggest above, I think the rowID in this test is more than just rowID?

Reply:
> Just to understand what's going on here: the rowID here is actually the full encoded cursor with timestamp and "real" rowID? So server side we end up overwriting the time boundaries from it?

Yes, e.g. `0cs1n52vmf2f4-06580c6bde001730-0001`. Just taken from the previous query result.